yarn_api_client/hadoop_conf.py

```diff
@@ -62,34 +57,20 @@ def get_resource_manager_host_port():
 def get_jobhistory_host_port():
     config_path = os.path.join(CONF_DIR, 'mapred-site.xml')
     prop_name = 'mapreduce.jobhistory.webapp.address'
-    value = parse(config_path, prop_name)
-    if value is not None:
-        host, _, port = value.partition(':')
-        return host, port
-    else:
-        return None
+    return parse(config_path, prop_name)
 
 
 def get_nodemanager_host_port():
     config_path = os.path.join(CONF_DIR, 'yarn-site.xml')
     prop_name = 'yarn.nodemanager.webapp.address'
-    value = parse(config_path, prop_name)
-    if value is not None:
-        host, _, port = value.partition(':')
-        return host, port
-    else:
-        return None
+    return parse(config_path, prop_name)
 
 
-def get_webproxy_host_port():
+def get_webproxy_host_port(timeout=30, auth=None, verify=True):
```

Comment:
I was thinking about switching test for RM to requests and utilize Auth/verify. Mainly because I'm not about behaviour of httplib when user is using some broken self-signed certificate. Same thing - will RM check work in kerberized environment if Auth is not passed?

Reply:
I don't know web protocols well enough to answer that question, but it seems like we shouldn't add these parameters until they're needed.  It looks like `check_is_active_rm()` is merely a ping against the `/cluster` endpoint, so I suspect the current code is sufficient.  If this is changed to use `requests`, then it does seem like `auth` and `verify` should probably be provided - but, until then, I would vote to remove unused parameters.

Reply:
Sure, no problem. Adjusted.